### ISSUE TITLE
triangulation error with nested MultiPolygons

### ISSUE THREAD

**Issue by orestes13** [orestes13/OneFineMesh#1]
During the calculation of the union of triangles whose circumcircle contains the point to be added, an error can occur if the union contains nested MultiPolygons.
```bash
Traceback (most recent call last):
  File "__init__.py", line 428, in <module>
    triangles = new_triangulate(points)
  File "__init__.py", line 333, in new_triangulate
    for i, x in enumerate(pol.boundary.coords[:-1]):
  File "/usr/lib64/python2.7/site-packages/shapely/geometry/base.py", line 741, in coords
    "Multi-part geometries do not provide a coordinate sequence")
NotImplementedError: Multi-part geometries do not provide a coordinate sequence
```

**Comment by orestes13** [comment]
Code to catch multipolygons included in 8209d34.

**Comment by orestes13** [comment]
The containing union of triangles (the cavity) can be either a Polygon, or MultiPolygon. The latter tends to occur when we are removing several ghost triangles. This issue really relates to an error when the boundary of the cavity is a MultiLineString (which it should not be).

This error occurs due to triangles not being correctly removed, and so there are triangles intersecting each other in the cavity. This is a consequence of precision issues when determining whether a triangle's circumcircle contains a point, for a point near the edge.

To alleviate this issue, added some tolerance checking to the point_in_circle function.
